compiler/rustc_codegen_llvm/src/base.rs

```diff
@@ -47,6 +46,21 @@ pub fn write_compressed_metadata<'tcx>(
     use snap::write::FrameEncoder;
     use std::io::Write;
 
+    // Historical note:
+    //
+    // When using link.exe it was seen that the section name `.note.rustc`
+    // was getting shortened to `.note.ru`, and according to the PE and COFF
+    // specification:
+    //
+    // > Executable images do not use a string table and do not support
+    // > section names longer than 8 characters
+    //
+    // https://docs.microsoft.com/en-us/windows/win32/debug/pe-format
+    //
+    // As a result, we choose a slightly shorter name! As to why
+    // `.note.rustc` works on MinGW, that's another good question...
```

Comment:
You can find nice expalination at https://github.com/llvm/llvm-project/blob/7f78e409d0280c62209e1a7dc8c6d1409acc9184/lld/COFF/Writer.cpp#L1194-L1201

Reply:
How should I add that link to the source? It is longer than 100 characters and as such tidy would complain.

Reply:
That works fine!